cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractConductingEquipmentConversion.java

```diff
@@ -471,6 +618,135 @@ protected void addMappingForTopologicalNode(Identifiable<?> identifiable, int te
         addMappingForTopologicalNode(identifiable, cgmesTerminalNumber, iidmTerminalNumber);
     }
 
+    public BoundaryLine createBoundaryLine(String boundaryNode) {
+        int modelEnd = 1;
+        if (nodeId(1).equals(boundaryNode)) {
+            modelEnd = 2;
+        }
+        String id = iidmId();
+        String name = iidmName();
+        String modelIidmVoltageLevelId = iidmVoltageLevelId(modelEnd);
+        boolean modelTconnected = terminalConnected(modelEnd);
+        String modelBus = busId(modelEnd);
+        String modelTerminalId = terminalId(modelEnd);
+        String boundaryTerminalId = terminalId(modelEnd == 1 ? 2 : 1);
+        int modelNode = -1;
+        if (context.nodeBreaker()) {
+            modelNode = iidmNode(modelEnd);
+        }
+        PowerFlow modelPowerFlow = powerFlow(modelEnd);
+        return new BoundaryLine(id, name, modelIidmVoltageLevelId, modelBus, modelTconnected, modelNode,
+            modelTerminalId, boundaryTerminalId, modelPowerFlow);
+    }
+
+    public static class BoundaryLine {
```

Comment:
Same remark, I think you can make it an external class

Reply:
Done